### ISSUE TITLE
SNOW-889806: Timezone is discarded when reading from Parquet files

### ISSUE THREAD

**Issue by dvorst** [snowflakedb/snowflake-connector-python#1688]
### Python version

3.11.3

### Operating system and processor architecture

macOS-13.5-arm64-arm-64bit

### Installed packages

```python
python = "^3.11"
snowflake-connector-python = {extras = ["pandas"], version = "^3.1.0"}
pandas = "^2.0.3"
pytz = "^2023.3"
python-dotenv = "^1.0.0"
```

### What did you do?

Note: this is related to issue https://github.com/snowflakedb/snowflake-connector-python/issues/1687

A parquet file converts datetimes to UTC and saves them without timezone in the data fields. The parquet file does, however, store the timezone in its schema. When Snowflake reads from a parquet file it only seems to read the data fields and does not use the timezone indicated in the schema. Snowflake then casts the TIMESTAMP_NTZ to TIMESTAMP_TZ using the sessions timezone.

As an example:
If the pandas dataframe contains datetime: 
    2023-08-09 21:42:49.166503+02:00
then the parquet data field is: 
    [[2023-08-09 19:42:49.166503]]
and the parquet file schema is: 
    DT: timestamp[us, tz=Europe/Amsterdam]
However Snowflake only reads: 
    2023-08-09 19:42:49.166503
Assume for a moment that the session timezone is : 
    +08:00
Then snowflake casts the data from parquet to: 
    2023-08-09 19:42:49.166503+08:00

The code below does the following:
- Creates a pandas dataframe with datetime that has the timezone of Amsterdam
- Saves the dataframe as parquet
- Creates a Snowflake table and stage
- Puts the parquet file into the stage
- Changes the timezone to Asia to indicate that the sessions timezone is used.
- Copies the data from the staged parquet file into the table
- Reads the values from the table

At multiple stages some information is printed.

```python
from pathlib import Path
import pandas as pd
import pyarrow.parquet as pq
from datetime import datetime
from snowflake import connector
import pytz
import os
from dotenv import load_dotenv

load_dotenv(override=True)


# snowflake connect configurations
snowflake_config = {
    "account": os.getenv("SF_ACCOUNT"),
    "user": os.getenv("SF_USER"),
    "role": os.getenv("SF_ROLE"),
    "password": os.getenv("SF_PASSWORD"),
    "warehouse": os.getenv("SF_WAREHOUSE"),
    "database": os.getenv("SF_DATABASE"),
    "schema": schema,
}

# create pandas dataframe, save it as parquet and print its value
pth = str(Path(f"tmp.parquet").absolute())
df = pd.DataFrame({"DT": [datetime.now(tz=pytz.timezone("Europe/Amsterdam"))]})
df.to_parquet(pth)
print(df.DT[0], " original datetime")

with connector.connect(**snowflake_config) as con:
    # create table
    r = (
        con.cursor()
        .execute("create or replace temporary table tmp1 (dt timestamp_tz)")
        .fetchall()
    )
    # create temp stage
    r = (
        con.cursor()
        .execute("CREATE or replace STAGE tmp_stage file_format=(type=PARQUET)")
        .fetchall()
    )
    # put parquet file into stage
    r = con.cursor().execute(f"PUT 'file://{pth}' @tmp_stage").fetchall()
    # read the staged file
    r = con.cursor().execute(f"""select * from @tmp_stage""").fetchall()
    print(
        "The following is read from the staged file, note how it does not contain a timezone:\n",
        r[0][0],
    )
    # change the timezone to Asia
    r = (
        con.cursor()
        .execute("""ALTER SESSION SET TIMEZONE = 'Asia/Shanghai'""")
        .fetchall()
    )
    r = con.cursor().execute("""select CURRENT_TIMESTAMP()""").fetchall()
    print(r[0][0], " session timezone is changed to Changhai to show that Snowflake just casts the variable to the session timezone")

    # copy into table
    r = (
        con.cursor()
        .execute(
            """copy into tmp1 (dt) from (select $1:"DT" from @tmp_stage) file_format=(type=PARQUET)"""
        )
        .fetchall()
    )
    # read table
    r = con.cursor().execute(f"""select DT from tmp1""").fetchall()
    print(r[0][0], " datetime read from table, it contains the TIMESTAMP_NTZ value + session timezone")
    
f = pq.ParquetFile(pth)

print('-'*150)
print('Schema of the parquet file:')
print(f.schema)
print('-'*150)
print("Data of the parquet file:")
print(pq.ParquetFile(pth).read_row_group(0))
```

### What did you expect to see?

The code outputs the following:
```


2023-08-09 21:42:49.166503+02:00  original datetime
The following is read from the staged file, note how it does not contain a timezone:
 {
  "DT": "2023-08-09 19:42:49.166"
}
2023-08-10 03:42:52.319000+08:00  session timezone is changed to Changhai to show that Snowflake just casts the variable to the session timezone
2023-08-09 19:42:49.166503+08:00  datetime read from table, it contains the TIMESTAMP_NTZ value + session timezone

------------------------------------------------------------------------------------------------------------------------------------------------------
Schema of the parquet file:
<pyarrow._parquet.ParquetSchema object at 0x1154bab80>
required group field_id=-1 schema {
  optional int64 field_id=-1 DT (Timestamp(isAdjustedToUTC=true, timeUnit=microseconds, is_from_converted_type=false, force_set_converted_type=false));
}

------------------------------------------------------------------------------------------------------------------------------------------------------
Data of the parquet file:
pyarrow.Table
DT: timestamp[us, tz=Europe/Amsterdam]
----
DT: [[2023-08-09 19:42:49.166503]]
```

### Can you set logging to DEBUG and collect the logs?

```bash
N/A
```

**Comment by sfc-gh-sfan** [comment]
Hey, this is a known issue of parquet file on Snowflake and a fix is in progress (current estimation is 7.28 release, going out next week). Once it is released, you could add `use_logical_type` to your copy command to get the fixed behavior.

**Comment by sfc-gh-achandrasekaran** [comment]
This has since been released on the server side.
